WireAPI/Sources/WireAPI/APIs/ClientAPI/ClientAPIV0.swift

```diff
@@ -0,0 +1,175 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+class ClientAPIV0: ClientAPI, VersionedAPI {
+
+    let httpClient: any HTTPClient
+
+    init(httpClient: any HTTPClient) {
+        self.httpClient = httpClient
+    }
+
+    var apiVersion: APIVersion {
+        .v0
+    }
+
+    func getSelfClients() async throws -> [UserClient] {
+        let request = HTTPRequest(
+            path: "\(pathPrefix)/clients",
+            method: .get
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: .ok, type: ListUserClientV0.self)
+            .parse(response)
+    }
+
+    func getClients(for userIDs: Set<UserID>) async throws -> [UserClients] {
+        let body = try JSONEncoder.defaultEncoder.encode(UserClientsRequestV0(qualifiedIDs: Array(userIDs)))
+        let request = HTTPRequest(
+            path: "/users/list-clients/v2", /// v2 suffix required for api version v0 and v1, suffix removed from next versions
+            method: .post,
+            body: body
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: .ok, type: UserClientsV0.self)
+            .parse(response)
+    }
+}
+
+struct ListUserClientV0: Decodable, ToAPIModelConvertible {
```

Comment:
Unfortunately, no, we could have done as follows but the compiler doesn't allow to do so since we already have a conformance to APIModelConvertible from an Array<Element> in some other place. That is strange though, I would expect it to understand that this is a different Element type.

```
Conflicting conformance of 'Array<Element>' to protocol 'ToAPIModelConvertible'; there cannot be more than one conformance, even with different conditional bounds
```

```
extension [UserClientV0]: ToAPIModelConvertible {
    func toAPIModel() -> [UserClient] {
        map { $0.toAPIModel() }
    }
}
```

```
        return try ResponseParser()
            .success(code: .ok, type: [UserClientV0].self)
            .parse(code: response.statusCode, data: data)
```